res/kiwixAPI.js

```diff
@@ -0,0 +1,43 @@
+import "core-js/stable";
+import "regenerator-runtime/runtime";
+
+let zim = {toc: {}};
+
+function getSectionList(){
+    let sectionList = [];
+    for (i = 0; i < document.querySelectorAll('h1, h2, h3, h4, h5, h6').length; i++) {
+        let headerObject = document.querySelectorAll('h1, h2, h3, h4, h5, h6')[i];
+        if (headerObject.id === "") {
+            headerObject.id = "documentparserid" + i;
+        }
+        sectionList.push(headerObject);
+    }
+    return sectionList;
+}
+
+zim.toc.hasTableOfContent = function() {
+    return getSectionList().length > 0 ? true : false;
+}
+
+zim.toc.getSections = function() {
+    let respArrOfSections = [];
+    const sectionList = getSectionList()
+    sectionList.forEach(section => {
+        respArrOfSections.push({
+            "toc_level": section.tagName,
+            "section_id": section.id,
+            "section_name": section.innerText,
+        })
+    });
+    return respArrOfSections;
+}
+
+zim.toc.scrollToSection = function(index){
+    const sectionList = getSectionList();
+    const sectionIdElem = document.getElementById(sectionList[index].id);
```

Comment:
We can probably generate the `id` to query (`"documentparserid" + index`) and directly use it to get the `sectionIdElem` (`const sectionIdElem = document.getElementById("documentparserid"+index);`)

This way we don't need to get the section list.

---

And by the way, `sectionList[index]` is already the elem returned by `document.getElementById(secttion[index].id)`.

Reply:
`const sectionIdElem = document.getElementById("documentparserid"+index)` won't work in this case as we only give the `documentparserid` in the cases where id is not already present.

So, I used `sectionList[index]`

Reply:
Indeed.
So we can simply do `const sectionIdElem = getSectionList()[index];` I think.